sunpy/coordinates/transformations.py

```diff
@@ -76,108 +63,155 @@ def _observers_are_equal(obs_1, obs_2, string_ok=False):
 # =============================================================================
 
 
-@frame_transform_graph.transform(FunctionTransform, HeliographicStonyhurst,
-                                 HeliographicCarrington)
+def _transform_obstime(frame, obstime):
+    """
+    Transform a frame to a new obstime using the appropriate loopback transformation
+    """
+    if frame.obstime is None:
+        raise ValueError("To perform this transformation the coordinate"
+                         " Frame needs an obstime Attribute")
+
+    # If obstime is None or the obstime matches, nothing needs to be done
+    if obstime is None or np.all(frame.obstime == obstime):
+        return frame
+
+    if isinstance(frame, BaseCoordinateFrame):
+        frame_class = frame.__class__
```

Comment:
I generally prefer `frame_class = type(frame)` but makes little difference.

Reply:
I can go with that